foundry/applications/hotbar.d.ts

```diff
@@ -1 +1,138 @@
-declare class Hotbar extends Application {}
+/**
+ * The global action bar displayed at the bottom of the game view.
+ * The Hotbar is a UI element at the bottom of the screen which contains Macros as interactive buttons.
+ * The Hotbar supports 5 pages of global macros which can be dragged and dropped to organize as you wish.
+ *
+ * Left clicking a Macro button triggers its effect.
+ * Right clicking the button displays a context menu of Macro options.
+ * The number keys 1 through 0 activate numbered hotbar slots.
+ * Pressing the delete key while hovering over a Macro will remove it from the bar.
+ *
+ * @see {@link Macros}
+ * @see {@link Macro}
+ */
+
+declare class Hotbar extends Application {
+  constructor(options: Application.Options);
+
+  /**
+   * The currently viewed macro page
+   * @defaultValue `1`
+   */
+  page: number;
+
+  /**
+   * The currently displayed set of macros
+   * @defaultValue `[]`
+   */
+  macros: Macro[];
+
+  /**
+   * Track collapsed state
+   * @defaultValue `false`
+   */
+  protected _collapsed: boolean;
+
+  /**
+   * Track which hotbar slot is the current hover target, if any
+   * @defaultValue `null`
+   */
+  protected _hover: number | null;
+
+  /**
+   * @override
+   * @defaultValue
+   * ```
+   * mergeObject(super.defaultOptions, {
+   *   id: "hotbar",
+   *   template: "templates/hud/hotbar.html",
+   *   popOut: false,
+   *   dragDrop: [{ dragSelector: ".macro-icon", dropSelector: "#macro-list" }]
+   * })
+   * ```
+   */
+  static get defaultOptions(): Application.Options;
+
+  /** @override */
+  getData(options?: Application.RenderOptions): { page: number; macros: Macro[]; barClass: 'collapsed' | '' };
+
+  /**
+   * Get the Array of Macro (or null) values that should be displayed on a numbered page of the bar
+   * @param page -
+   */
+  protected _getMacrosByPage(page: number): Macro[];
+
+  /**
+   * Collapse the Hotbar, minimizing its display.
+   * @returns A promise which resolves once the collapse animation completes
+   */
+  collapse(): Promise<boolean>;
+
+  /**
+   * Expand the Hotbar, displaying it normally.
+   * @returns A promise which resolves once the expand animation completes
+   */
+  expand(): Promise<boolean> | boolean;
```

Comment:
The `expand` function is not async in foundry so the return type reflects that here, but I dont think it is intended in foundry.

Reply:
If that's still the case in 0.8.0, it's probably worth filing a bug report with foundry itself.
EDIT: Just checked, it's still the same in 0.8.0. I'll file a bug report.

Reply:
https://gitlab.com/foundrynet/foundryvtt/-/issues/4745